### PR TITLE
Document extractParserSettings

### PR DESCRIPTION
**Pull request by jonas**
Resolves one point in #656

Note that there were some unrelated formatting changes when running `docs/test`.

### CODE REVIEW

**Comment by jonas** [comment]
PLS BUILD

**Comment by jonas** [comment]
It looks like the build fails because the PR validator skips validation of the docs directory.

```
[info] Skipping validation of [akka-http-marshallers-java], as PR does NOT affect this project...
[info] Skipping validation of [akka-http-marshallers-scala], as PR does NOT affect this project...
[info] Skipping validation of [akka-parsing], as PR does NOT affect this project...
[info] Skipping validation of [root], as PR does NOT affect this project...
[info] Skipping validation of [akka-http-core], as PR does NOT affect this project...
[info] Skipping validation of [akka-http-xml], as PR does NOT affect this project...
[info] Skipping validation of [akka-http-docs], as PR does NOT affect this project...
[info] Skipping validation of [akka-http-jackson], as PR does NOT affect this project...
[info] Skipping validation of [akka-http-testkit], as PR does NOT affect this project...
[info] Skipping validation of [akka-http], as PR does NOT affect this project...
[info] Skipping validation of [akka-http-spray-json], as PR does NOT affect this project...
[info] Skipping validation of [akka-http2-support], as PR does NOT affect this project...
[info] Skipping validation of [akka-http-tests], as PR does NOT affect this project...
[success] Total time: 2 s, completed Dec 22, 2016 6:30:10 PM
Build step 'Conditional step (single)' changed build result to SUCCESS
[pr-validator-akka-http] $ /bin/bash -xe /tmp/hudson7340913945419520604.sh
Recording test results
ERROR: Step ‘Publish JUnit test result report’ failed: No test report files were found. Configuration error?
```

**Comment by ktoso** [comment]
Oh, we just changed today to run the `validatePullRequest` task in validation. Seems that needs to be tweaked a bit

**Comment by jonas** [comment]
Ah, OK. I was looking to see if I could tweak this in `project/ValidatePullRequest.scala`.

**Comment by ktoso** [comment]
Yeap, behaviour would be best tweaked there.

**Comment by jonas** [comment]
PLS BUILD ALL

**Comment by jonas** [comment]
Not sure if Jenkins saw the 'all' build remark or the last commit actually worked.

**Comment by ktoso** [comment]
Yes, PLS BUILD ALL works :) Sorry for the misconfig, will need to fix that.

File: project/ValidatePullRequest.scala [hunk 1]
```diff
@@ -168,7 +169,7 @@ object ValidatePullRequest extends AutoPlugin {
           val dirtyDirectories = statusOutput
             .map(l ⇒ l.trim.dropWhile(_ != ' ').drop(1))
             .map(_.takeWhile(_ != '/'))
-            .filter(dir => dir.startsWith("akka-") || dir == "project")
+            .filter(dir => dir.startsWith("akka-") || dir.startsWith("docs") || dir == "project")
```
Comment (ktoso) [inline project/ValidatePullRequest.scala]: Great, thanks - this should fix the PR validation.

**Pull request merged**